fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/impl/FHIRPersistenceJDBCImpl.java

```diff
@@ -765,6 +739,201 @@ private ParameterDAO makeParameterDAO(Connection connection) throws FHIRPersiste
+        // results.
+        for (InclusionParameter includeParm : searchContext.getIncludeParameters()) {
+            if (!includeParm.isIterate()) {
+                // Build and run the query
+                List<com.ibm.fhir.persistence.jdbc.dto.Resource> includeResources =
+                        this.runIncludeQuery(resourceType, searchContext, queryBuilder, includeParm, SearchConstants.INCLUDE,
+                            baseLogicalResourceIds, queryResultMap, resourceDao, 1, allResourceIds);
+                
+                // Add new ids to de-dup list
+                allResourceIds.addAll(includeResources.stream().map(r -> r.getId()).collect(Collectors.toSet()));
+
+                // Add resources to list
+                allIncludeResources.addAll(includeResources);
+                
+                // Check if max size exceeded. If so, return results and let rest helper throw exception.
+                if (allIncludeResources.size() > SearchConstants.MAX_PAGE_SIZE) {
+                    return allIncludeResources;
+                }
+            }
+        }
+
+        // Process non-iterative _revinclude parameters. These are only run against 'match' search
+        // results.
+        for (InclusionParameter revincludeParm : searchContext.getRevIncludeParameters()) {
+            if (!revincludeParm.isIterate()) {
+                // Build and run the query
+                List<com.ibm.fhir.persistence.jdbc.dto.Resource> revincludeResources =
+                        this.runIncludeQuery(resourceType, searchContext, queryBuilder, revincludeParm, SearchConstants.REVINCLUDE,
+                            baseLogicalResourceIds, queryResultMap, resourceDao, 1, allResourceIds);
+
+                // Add new ids to de-dup list
+                allResourceIds.addAll(revincludeResources.stream().map(r -> r.getId()).collect(Collectors.toSet()));
+
+                // Add resources to list
+                allIncludeResources.addAll(revincludeResources);
+                
+                // Check if max size exceeded. If so, return results and let rest helper throw exception.
+                if (allIncludeResources.size() > SearchConstants.MAX_PAGE_SIZE) {
+                    return allIncludeResources;
+                }
+            }
+        }
+        
+        // Process iterative parameters.
+        // - Iteration 0 is a special iteration. It will only process against resources returned by primary search
+        //   if the iterative parameter's target type is the same as the primary search resource type.
+        // - Iteration 1 processes against resources returned by primary search or by non-iterative
+        //   _include and _revinclude search.
+        // - Iteration 2 and above processes only against resources returned by the previous iteration. Note
+        //   that we currently have a max of only one iteration.
```

Comment:
I'm assuming this means Iteration 0 and Iteration 1, so that's two iterations. Might be worth clarifying in the comment. Why are we limiting this?

Reply:
Comment updated.